src/coreclr/src/jit/gentree.h

```diff
@@ -3951,7 +3951,11 @@ struct GenTreeCall final : public GenTree
     CORINFO_SIG_INFO* callSig;
 #endif
 
-    TailCallSiteInfo* tailCallInfo;
+    union {
+        TailCallSiteInfo* tailCallInfo;
+        // Only used for unmanaged calls, which cannot be tail-called
+        CorInfoUnmanagedCallConv unmgdCallConv;
+    };
```

Comment:
Hate to see even more unioning in call nodes. I wonder if we have enough free space elsewhere in the node to introduce a tag for these without increasing overall size.

Or better yet figure out the right extensibility mechanism and start using that.

Reply:
On this topic, I noticed recently that there is prior art:
https://github.com/dotnet/runtime/blob/202f6e1deede57ad127445256cd1d5c2bbad136f/src/coreclr/src/jit/compiler.h#L10113-L10157
Maybe it would suffice to make the value of this table a linked list of annotations, or introduce a new table similar to this.

Reply:
Perhaps? 

I worry that this sort of "side table" can introduce latent maintenance issues. But perhaps this is not too bad as the JIT doesn't recycle nodes and (hopefully) bashes nodes from one type to another in fairly systematic ways, so there's just a few places to where invalidation might be needed.